pkg/plugin/eventqueueset.go

```diff
@@ -26,9 +36,21 @@ func (s eventQueueSet[T]) enqueue(key string, item T) error {
 	hash := hasher.Sum64()
 
 	idx := int(hash % uint64(len(s.queues)))
-	return s.queues[idx].Add(item)
+
+	s.metrics.eventQueueDepth.Inc()
```

Comment:
We have multiple queues here, and there is a potential situation where only one of those gets very long if the problem is caused by one pod/node.  We won't see this if we only have a single metric for all the queues.

Wdyt about exposing 1 metric per queue or a histogram? This increases cardinality, so it is up for debate.

Reply:
I see your point, but and the same time I'm not sure if that's really needed. We have like 64 workers and I'm not sure visualizing them all would be useful at this point. I suggest starting off with this simpler version and then add metrics for each worker if needed.

Reply:
I think a reasonable middle ground might be to include the worker number in the logs but keep the metrics aggregated. That way, if we see issues, it's still possible to assess the distribution of impact, but we don't end up creating thousands of time series :sweat_smile:

Reply:
We already include pod names in the logs, so if there is a problem with a specific group of pods, it is already visible in the logs. I don't see how logging the queue index could help here. To troubleshot any issue, we have to look for the pod name, not the queue index. Plus, if a particular queue is acting slow, the p99 of latency metrics would rise so we can get notified and see the logs.

Reply:
The idea behind logging the queue index is that it becomes possible to determine the cause of widespread impact. Idea being that this:

> if a particular queue is acting slow, the p99 of latency metrics would rise so we can get notified and see the logs

works as a start, but doesn't confirm anything because it's very difficult to see "oh, all these events with high latency came through the same queue"

Basically: while the problem may already be visible, the cause may not.

Reply:
> works as a start, but doesn't confirm anything because it's very difficult to see "oh, all these events with high latency came through the same queue"

My point is knowing that all high latency event came through the same queue doesn't help us with debugging. Because whenever we see eventqueue has a high latency, we would look for pods/VMs  with errors in the logs. Knowing the associated queue index here doesn't give us any useful information.

I'm just trying to be conservative with what we log to keep the logs concise and useful, otherwise, it's fine to log the queue index here.